src/ZkFold/P2P/Contract.hs

```diff
@@ -168,15 +172,13 @@ p2pMatchedOrderContract
     => ScalarField (Ed25519 a) ~ UInt 256 a
     => BaseField (Ed25519 a) ~ UInt 256 a
     => Point (Ed25519 a)
-    -> Transaction inputs rinputs  outputs tokens () a
+    -> Transaction inputs rinputs outputs tokens () a
     -> MatchedOffer a
     -> Bool a
 p2pMatchedOrderContract vk tx mo@(MatchedOffer (addr, transfer, signature)) =
-    let h                  = hashMatchedOffer mo
-        f = (\o acc -> bool @(Bool a) acc (Just o) (txoDatumHash o == h))
+    let h = hashMatchedOffer mo
+        f o acc = bool @(Bool a) acc (Just o) (txoDatumHash o == h)
         -- TODO: Simplify this using symbolic `find`.
-        v = (\(Output (_, (v', _))) -> v') $ fromJust $ foldr f Nothing $
-            fmap (\(Input (_, o)) -> o) $ txInputs tx
-        -- TODO: Instead of `zero`, it should be the hash of `()`.
-        txo                = Output (addr, (v, fromConstant (0 :: Natural) :: ByteString 256 a)) :: Output tokens () a
+        v = txoDatumHash $ fromJust $ foldr f Nothing $ txiOutput <$> txInputs tx
+        txo = Output (addr, (emptyValue, v :: ByteString 256 a)) :: Output tokens () a
```

Comment:
A transaction output cannot have an "empty" value. This is not a correct modification. Instead, the datum must be the hash of `()`, as it was in the TODO.

Reply:
@vlasin Yes, I was confused after rebasing since @vks4git already solved that issue.

Reply:
Well, not completely. As it says, the value of the datum should be the hash of `()`. You can probably ask @zlonast for the exact value.

Reply:
@zlonast I mentioned you because you should probably know the hash of `()`. I believe it is not the same as a hash of an empty string because everything gets converted to `BuiltinData`, then to bytes, and then hashed, right?

Reply:
@vlasin I think is just a `mempty` cborg. 

```
instance Era era => EncCBOR (Datum era) where
   ...
   NoDatum -> OmitC NoDatum
```
https://github.com/IntersectMBO/cardano-ledger/blob/b3b8b3d34742374319e7afef236ed1d29effd6b9/libs/cardano-ledger-core/src/Cardano/Ledger/Plutus/Data.hs#L223

```
  -- | Skip over the  (component,field, argument), don't encode it at all (used in sparse encoding).
  OmitC :: t -> Encode w t
```

```
encodeCountPrefix :: Word -> Encode w t -> Encoding
encodeCountPrefix _ (OmitC _) = mempty
```

```
gsize :: Encode w t -> Word
gsize (OmitC _) = 0
```
https://github.com/IntersectMBO/cardano-ledger/blob/e30e474370f4c3ca55a0a67b4e71bab7aca22a26/libs/cardano-ledger-binary/src/Cardano/Ledger/Binary/Encoding/Coders.hs#L197C37-L197C43